src/document/advanced.rs

```diff
@@ -70,33 +72,41 @@ impl DocumentAdvancedOps for crate::IronOxide {
                 }
             };
 
-        internal::document_api::encrypted_document_unmanaged(
-            self.device.auth(),
-            &self.recrypt,
-            &self.user_master_pub_key,
-            &self.rng,
-            data,
-            encrypt_opts.id.clone(),
-            grant_to_author,
-            &explicit_users,
-            &explicit_groups,
-            policy_grants,
+        run_maybe_timed_sdk_op(
+            internal::document_api::encrypted_document_unmanaged(
```

Comment:
Not part of the work you're doing, but I'm surprised that `encrypted_document_unmanaged()` is that function's name

Reply:
Bad name, indeed. It's internal, so easily changed. New name will be `encrypt_document_unmanaged`